src/omnicore/rpctxobject.cpp

```diff
@@ -256,6 +256,7 @@ void populateRPCTypeTradeOffer(CMPTransaction& omniObj, Object& txobj)
 
     // Check levelDB to see if the amount for sale has been amended due to a partial purchase
     // TODO: DEx phase 1 really needs an overhaul to work like MetaDEx with original amounts for sale and amounts remaining etc
+    // TODO: if the amount of the transaction is not used here, then the BTC amount should be recalculated (?)
```

Comment:
I agree, this would probably be the way to go.

However, there is also a difference between "original amounts" and "the amounts of the transaction".

In the example above the "transaction amounts" are 1000.0/1000.0, while the "original amounts" are 50.0/50.0 (because only that much balance exists), whereby I mean with "original amount" the "amount initially up for sale".